### PR TITLE
[TEST] update to cb3 using `conda smithy update-cb3`

### PR DESCRIPTION
**Pull request by shadowwalkersb**

### CODE REVIEW

**Comment by shadowwalkersb** [comment]
The goal of this PR was to test `conda smithy update-cb3`, but as a successful result, we have also updated the recipe. Should we just update the title and merge?

File: .appveyor.yml [hunk 1]
```diff
@@ -9,29 +9,8 @@ environment:
...
-
-    - TARGET_ARCH: x86
-      CONDA_PY: 36
-      CONDA_INSTALL_LOCN: C:\\Miniconda36
-
-    - TARGET_ARCH: x64
-      CONDA_PY: 36
-      CONDA_INSTALL_LOCN: C:\\Miniconda36-x64
+    - CONFIG: win_
+      CONDA_INSTALL_LOCN: C:\Miniconda36-x64
```
Comment (jakirkham) [inline .appveyor.yml]: Hmm...seems to have dropped the VS 2008 builds. 😕

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
+          command: ./.circleci/run_docker_build.sh
 
 workflows:
   version: 2
   build_and_test:
     jobs:
-      - build
+      - build_linux_python2.7
+      - build_linux_python3.5
+      - build_linux_python3.6
```
Comment (jakirkham) [inline .circleci/config.yml]: Not sure why we have a Python matrix on Linux.

File: .travis.yml [hunk 3]
```diff
@@ -7,6 +7,11 @@ os: osx
 osx_image: xcode6.4
 
 env:
+  matrix:
+    - CONFIG=osx_python2.7
+    - CONFIG=osx_python3.5
+    - CONFIG=osx_python3.6
```
Comment (jakirkham) [inline .travis.yml]: ...or on macOS for that matter.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -10,26 +10,19 @@ source:
...
 
 requirements:
   build:
-    - python  # [win]
-    # This breals a dependency cycle:
+    - {{ compiler('c') }}
+  host:
+    # This breaks a dependency cycle:
     # curl->libssh2->cmake->curl
     - cmake-binary
```
Comment (jakirkham) [inline recipe/meta.yaml]: This can be moved to `build`.

Comment (jakirkham) [reply recipe/meta.yaml]: Maybe we can switch to `cmake` afterwards as well. Thoughts from others?

**Comment by jakirkham** [comment]
@conda-forge-admin, please re-render.

Edit: Hopefully fixes the weird Python matrices and missing VS 2008 build.

**Comment by shadowwalkersb** [comment]
Yeah, forgot to rerender. Was focusing on the update-cb3 stuff.

File: recipe/meta.yaml (hunk 4 above)
Comment (shadowwalkersb) [reply recipe/meta.yaml]: > Maybe we can switch to cmake afterwards as well. Thoughts from others?

Tried that and got something like this.

```
+ conda build /home/conda/recipe_root -m /home/conda/feedstock_root/.ci_support/linux_.yaml --quiet
BUILD START: ['libssh2-1.8.0-h5b517e9_2.tar.bz2']
libssh2 cannot depend on itself
Exited with code 1
```

Comment (shadowwalkersb) [reply recipe/meta.yaml]: > This can be moved to build.

Done.

Comment (jakirkham) [reply recipe/meta.yaml]: Was that when it was in `host` or after moving it to `build`?

**Comment by jakirkham** [comment]
No worries. Just wanted to make sure something weird wasn't happening.

File: recipe/meta.yaml (hunk 4 above)
Comment (shadowwalkersb) [reply recipe/meta.yaml]: Can't track it, so maybe, better just try it here.

**Comment by shadowwalkersb** [comment]
`cmake` still causes the cyclic dependency.

```
+ conda build /home/conda/recipe_root -m /home/conda/feedstock_root/.ci_support/linux_.yaml --quiet
BUILD START: ['libssh2-1.8.0-h5b517e9_2.tar.bz2']
libssh2 cannot depend on itself
Exited with code 1
```

**Comment by jakirkham** [comment]
Thanks for trying. Have raised issue ( https://github.com/conda/conda-build/issues/2960 ) on this point.

**Comment by shadowwalkersb** [comment]
Anything else to do here? Should we merge?

**Comment by shadowwalkersb** [comment]
ping

**Pull request closed without merging**